### PR TITLE
Improve logging

### LINKED ISSUE #428: Improve logging
- Check that everything goes to file
 - Add more info messages as well
 - Check that all promises are handled, especially regarding BN and validators db

### PR DESCRIPTION
**Pull request by BeroBurny**
closes #428

### CODE REVIEW

File: src/main/gui/utils.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
 export async function installExtensions(): Promise<void | string[]> {
     // eslint-disable-next-line @typescript-eslint/no-require-imports,@typescript-eslint/no-var-requires
     const installer = require("electron-devtools-installer");
     const forceDownload = !!process.env.UPGRADE_EXTENSIONS;
     const extensions = ["REACT_DEVELOPER_TOOLS", "REDUX_DEVTOOLS"];
 
-    return Promise.all(extensions.map((name) => installer.default(installer[name], forceDownload))).catch(logger.log);
+    return Promise.all(extensions.map((name) => installer.default(installer[name], forceDownload))).catch(
+        mainLogger.log,
```
Comment (mpetrunic) [inline src/main/gui/utils.ts:10]: error?

File: src/main/logger/index.ts [hunk 2]
```diff
@@ -1 +1,21 @@
-export * from "./lodestar";
+import electronLog from "electron-log";
```
Comment (mpetrunic) [inline src/main/logger/index.ts:1]: maybe use named import?

File: src/renderer/ducks/auth/sagas.ts [hunk 3]
```diff
@@ -2,20 +2,23 @@ import {all, takeEvery, put, PutEffect} from "redux-saga/effects";
...
-export function* authorize(): Generator<Promise<CGAccount> | PutEffect, void, CGAccount & null> {
+export function* authorize(): Generator<Promise<CGAccount> | PutEffect, void, CGAccount | null> {
     try {
         const account = yield database.account.get(DEFAULT_ACCOUNT);
         if (account !== null) {
+            chainGuardianLogger.info("Found Account with", account.getValidatorsAddresses().length, "validator/s");
             yield put(storeAuth(account));
             yield put(loadValidatorsAction());
+        } else {
+            chainGuardianLogger.info("Account not found");
```
Comment (mpetrunic) [inline src/renderer/ducks/auth/sagas.ts:18]: warn?

File: src/main/logger/index.ts [hunk 4]
```diff
@@ -1 +1,21 @@
...
+    logger.transports.file.fileName = file;
+
+    if (process.env.NODE_ENV === "production") logger.transports.console.level = false;
+
+    return logger;
+};
+
+export const mainLogger = createLogger("mainLogger", "core.log");
+
+export const chainGuardianLogger = createLogger("chainGuardian", "chainGuardian.log");
```
Comment (mpetrunic) [inline src/main/logger/index.ts:14]: maybe some shorter names for usage like. mainLog, appLog or something?

File: src/renderer/ducks/auth/sagas.ts (hunk 3 above)
Comment (BeroBurny) [reply src/renderer/ducks/auth/sagas.ts:18]: I don't think this should be a warning this is only the case when the user dint import any validator

File: src/main/logger/index.ts (hunk 4 above)
Comment (BeroBurny) [reply src/main/logger/index.ts:14]: keeping a short as `log` can be confusing but can change `chainGuardianLogger` to `cgLogger`

**Review by mpetrunic — APPROVED** [review]

**Pull request merged**